### PR TITLE
[Core] Restart process: Naming update to io_path

### PR DESCRIPTION
**Pull request by mpentek**
Update io_foldername to io_path, consistent with recent changes in the ascii file  writer and vtk output.

Test pass, example case in my machine passes in OpenMP and MPI.

### CODE REVIEW

**Comment by sunethwarna** [comment]
I am ok with the change :)... Thanks.... :) I think this need to be approved by technical commitee ryt?

**Comment by philbucher** [comment]
I will post sth tmr please wait with merging

File: kratos/python_scripts/save_restart_process.py [hunk 1]
```diff
@@ -17,19 +17,22 @@ def __init__(self, model, params):
...
         default_settings = KratosMultiphysics.Parameters("""{
-            "help"                         : "This process is used in order to save the problem databse with the serializer the current problem",
             "model_part_name"              : "SPECIFY_MODEL_PART_NAME",
             "echo_level"                   : 0,
             "serializer_trace"             : "no_trace",
             "restart_save_frequency"       : 0.0,
             "restart_control_type"         : "time",
             "save_restart_files_in_folder" : true,
-            "io_foldername"                : ""
+            "io_path"                : ""
```
Comment (philbucher) [inline kratos/python_scripts/save_restart_process.py]:
```suggestion
            "output_path"                : ""
```

this is what we started doing for the other processes

File: kratos/python_scripts/restart_utility.py [hunk 2]
```diff
@@ -18,7 +18,7 @@ def __init__(self, model_part, settings):
         default_settings = KratosMultiphysics.Parameters("""
         {
             "input_filename"                 : "",
-            "io_foldername"                  : "",
+            "io_path"                        : "",
```
Comment (philbucher) [inline kratos/python_scripts/restart_utility.py]: here I guess the name is ok, although it is again not consistent :/

File: kratos/python_scripts/save_restart_process.py (hunk 1 above)
Comment (mpentek) [reply kratos/python_scripts/save_restart_process.py]: My objection is: restart uses that path to output and input, thus the naming io. This is something that differentiates it from other processes, which only output data.

Comment (philbucher) [reply kratos/python_scripts/save_restart_process.py]: this is the save_restart process, this one only outputs

File: kratos/python_scripts/restart_utility.py (hunk 2 above)
Comment (mpentek) [reply kratos/python_scripts/restart_utility.py]: See above, but we could have the output_path for the save_restart and the io_path for the general restart_utility. I would be in favor of io_path for both, but the main importance is we change from folder_name as that is confusing.

Or we switch here straight to input_path? As it is internally used by load_restart. Once we agree I need to update and test.

Comment (philbucher) [reply kratos/python_scripts/restart_utility.py]: as I said above I would like it to be 100% consistent, at least for the output processes (which includes the restart process)

the restart utility is used differently, hence there I would accept a different name

For me as it is now is ok

Comment (mpentek) [reply kratos/python_scripts/restart_utility.py]: For the RestartUtility the PathLoad and PathSave use the same, so I will revert to using output_path.

**Review by philbucher — APPROVED** [review]
:+1:

**Pull request merged**